base/libgit2/reference.jl

```diff
@@ -166,6 +166,9 @@ then `ref` will be peeled until an object other than a `GitTag` is obtained.
 
 - A `GitTag` will be peeled to the object it references.
 - A `GitCommit` will be peeled to a `GitTree`.
+
+Note that only annotated tags can be peeled to `GitTag` objects. Lightweight tags (the
```

Comment:
Nitpick but should we make this a `!!! note`? Also, do you want to have x-refs here for the types?